### PR TITLE
[frontend] speed up request creation

### PR DESCRIPTION
**Pull request by adrianschroeter**
by avoiding generating large diff without any consumer.

The frontend should not overwrite backend defaults in general.
It may make sense to do it for specific views.

### CODE REVIEW

File: src/api/app/models/bs_request_action.rb [hunk 1]
```diff
@@ -238,7 +238,7 @@ def notify_params(ret = {})
   end
 
   def contains_change?
-    return sourcediff.present?
+    return sourcediff(nodiff: 1).present?
```
Comment (hennevogel) [inline src/api/app/models/bs_request_action.rb:241]: What does `nodiff` do? If this suppresses the diff the `.present?` will not return the opposite of what it returned before.

**Comment by adrianschroeter** [comment]
On Donnerstag, 16. April 2020, 19:23:27 CEST wrote Henne Vogelsang:
> @hennevogel commented on this pull request.
> 
> 
> 
> > @@ -238,7 +238,7 @@ def notify_params(ret = {})
>    end
>  
>    def contains_change?
> -    return sourcediff.present?
> +    return sourcediff(nodiff: 1).present?
> 
> What does `nodiff` do? If this suppresses the diff the `.present?` will not return the opposite of what it returned before.

the code avoids generating large diffs, extracting tar balls and so on.

Esp. due the non-default very large limits in the frontend code this is slowing
down the call a lot. And why just need a binary yes/no answer here, so it is okay
to know if a tar ball got changed or not, but the actual change is unimportant.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by hennevogel** [comment]
To answer my own question: `nodiff` generates the `+++` & `---` lines of the unified diff output but not the lines in between. So only this.

```diff
--- /path/to/original	timestamp
+++ /path/to/new	timestamp
```

So `.present?` will be fine.

**Comment by vpereira** [comment]
it looks like some VCR files shoud be regenerated.

**Comment by hennevogel** [comment]
> it looks like some VCR files shoud be regenerated.

some == _all_ that have anything to do with requests as `nodiff` is now default.

**Comment by hennevogel** [comment]
@adrianschroeter do you need help with regenerating the cassettes?

**Comment by adrianschroeter** [comment]
On Montag, 27. Juli 2020, 18:58:44 CEST wrote Henne Vogelsang:
> @adrianschroeter do you need help with regenerating the cassettes?

feel free to do it as always, thanks!

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Pull request merged**